src/lib/__tests__/index-test.ts

```diff
@@ -0,0 +1,855 @@
+
+        instance.addWidgets([createSearchBox()]);
+
+        instance.init!(
+          createInitOptions({
+            instantSearchInstance,
+          })
+        );
+
+        expect(instance.getHelper()!.state).toEqual(
+          algoliasearchHelper.SearchParameters.make({
+            index: 'index_name',
+            query: 'Apple',
+          })
+        );
+
+        instance.addWidgets([createPagination()]);
+
+        expect(instance.getHelper()!.state).toEqual(
+          algoliasearchHelper.SearchParameters.make({
+            index: 'index_name',
+            query: 'Apple',
+            page: 5,
+          })
+        );
+      });
+
+      it('calls `init` on the added widgets', () => {
+        const instance = index({ indexName: 'index_name' });
+        const instantSearchInstance = createInstantSearch();
+        const pagination = createSearchBox({
+          init: jest.fn(),
+        });
+
+        instance.init!(
+          createInitOptions({
+            instantSearchInstance,
+          })
+        );
+
+        expect(pagination.init).toHaveBeenCalledTimes(0);
+
+        instance.addWidgets([pagination]);
+
+        expect(pagination.init).toHaveBeenCalledTimes(1);
+        expect(pagination.init).toHaveBeenCalledWith({
+          instantSearchInstance,
+          parent: instance,
+          helper: instance.getHelper(),
+          state: instance.getHelper()!.state,
+          templatesConfig: instantSearchInstance.templatesConfig,
+          createURL: instantSearchInstance._createAbsoluteURL,
+          onHistoryChange: instantSearchInstance._onHistoryChange,
+        });
+      });
+
+      it('schedules a search to take the added widgets into account', () => {
+        const instance = index({ indexName: 'index_name' });
+        const instantSearchInstance = createInstantSearch({
+          scheduleSearch: jest.fn(),
+        });
+
+        instance.init!(
+          createInitOptions({
+            instantSearchInstance,
+          })
+        );
+
+        expect(instantSearchInstance.scheduleSearch).toHaveBeenCalledTimes(0);
+
+        instance.addWidgets([createSearchBox()]);
+
+        expect(instantSearchInstance.scheduleSearch).toHaveBeenCalledTimes(1);
+      });
+
+      it('does not trigger a search without widgets to add', () => {
+        const instance = index({ indexName: 'index_name' });
+        const instantSearchInstance = createInstantSearch({
+          scheduleSearch: jest.fn(),
+        });
+
+        instance.init!(
+          createInitOptions({
+            instantSearchInstance,
+          })
+        );
+
+        expect(instantSearchInstance.scheduleSearch).toHaveBeenCalledTimes(0);
+
+        instance.addWidgets([]);
+
+        expect(instantSearchInstance.scheduleSearch).toHaveBeenCalledTimes(0);
+      });
+    });
+  });
+
+  describe('removeWidgets', () => {
+    it('removes given widgets from the instance', () => {
```

Comment:
It's worth also adding a test removing only _some_ widgets (not all).

Reply:
d0fd1aa6d069a7d48f22976ed7825cbc2e886542